src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -633,7 +641,105 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await TryHandleTimeoutAsync(invokeTask, CancellationToken.None, throwOnTimeout, timeoutTokenSource.Token, timerInterval, instance, null);
             }
 
-            await invokeTask;
+            FunctionResult functionResult = null;
+
+            try
+            {
+                await invokeTask;
+                functionResult = new FunctionResult(true);
+            }
+            catch (Exception ex)
+            {
+                functionResult = new FunctionResult(ex);
+                throw;
+            }
+            finally
+            {
+                await InvokeExecutedFilters(invoker, invokeParameters, functionCancellationTokenSource, instance, logger, jobHost, config, functionResult, parameterNames);
+            }
+        }
+
+        private static async Task InvokeExecutingFilters(IFunctionInvoker invoker, object[] invokeParameters, CancellationTokenSource functionCancellationTokenSource,
+            IFunctionInstance instance, ILogger logger, JobHost jobHost, JobHostConfiguration config, IReadOnlyList<string> parameterNames)
+        {
+            MethodInfo functionMethod = instance.FunctionDescriptor.Method;
+            CancellationToken cancellationToken = functionCancellationTokenSource.Token;
+            var invokeFilters = functionMethod.GetCustomAttributes().OfType<InvocationFilterAttribute>();
+
+            var parameters = new Dictionary<string, object>();
+
+            for (var i = 0; i < invokeParameters.Length; i++)
+            {
+                parameters.Add(parameterNames.ToArray<string>()[i], invokeParameters[i]);
+            }
+
+            if (invokeFilters != null)
+            {
+                foreach (var filter in invokeFilters)
+                {
+                    try
+                    {
+                        if (filter.GetType() == typeof(InvokeFunctionFilterAttribute))
```

Comment:
Shouldn't need this special type casing here. Since the JobHost/Config properties are internal, I think it's fine for them to always be set, which would allow you to compress this code.